frontend/src/components/SaveDialog.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react'
+import {
+  Button,
+  Collapse,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  Grid,
+  Radio,
+  RadioGroup,
+  TextField,
+} from '@material-ui/core'
+
+interface Props {
+  open: boolean
+  handleClose: () => void
+  handleSubmit: (newBranch: boolean, name: string, message: string) => void
+  currentBranch: string
+}
+
+const SaveDialog = ({
+  open,
+  handleClose,
+  handleSubmit,
+  currentBranch,
+}: Props) => {
+  const [createNewBranch, setCreateNewBranch] = useState(false)
+  const [branchName, setBranchName] = useState('')
+  const [commitMessage, setCommitMessage] = useState('')
+
+  const handleExit = () => {
+    setCreateNewBranch(false)
+    setBranchName('')
+    setCommitMessage('')
+  }
+
+  return (
+    <Dialog
+      open={open}
+      onClose={handleClose}
+      onExit={handleExit}
+      aria-labelledby="form-dialog-title"
+    >
+      <DialogTitle id="form-dialog-title">Save changes</DialogTitle>
+      <DialogContent>
+        <Grid container direction="column">
+          <Grid item>
+            <FormControl component="fieldset">
+              <FormLabel component="legend">Select branch</FormLabel>
+              <RadioGroup
+                aria-label="branch selection"
+                value={createNewBranch}
+                onChange={(event) =>
+                  setCreateNewBranch(
+                    (event.target as HTMLInputElement).value === 'true'
```

Comment:
I think the event gets right type automatically without so ` as HTMLInputElement` is redundant.